### DISCUSSION TITLE
collection.insert如何避免相同id的资源被重复添加到向量数据库？

### DISCUSSION THREAD

**Discussion by jilieryuyi** [milvus-io/milvus#7224] [Q&A and General discussion]
求教，如题

**Comment by xiaofan-luan** [comment]
可以直接不指定id？如果目前需要自定义id的话可以先用kv数据库或者mysql存一下映射关系,milvus 2.0后续会加入主键去重功能，依赖一个外置的kv数据库去做去重，不过目前这个功能还没有。

**Reply by jilieryuyi** [reply]
也没找到根据id删除的api ~

**Reply by jilieryuyi** [reply]
> �可以直接不指定id？如果目前需要自定义id的话可以先用kv数据库或者mysql存一下映射关系,milvus 2.0后续会加入主键去重功能，依赖一个外置的kv数据库去做去重，不过目前这个功能还没有。

业务场景一般都需要id关联 ~ 所以会指定id

**Reply by xiaofan-luan** [reply]
嗯，删除功能目前正在开发中，社区有对应issue。

id去重目前可能需要外部系统一起解决，如果有什么方案可以贡献进社区！
